public/app/plugins/panel/flamegraph/components/FlameGraph/rendering.ts

```diff
@@ -8,10 +11,99 @@ import {
   LABEL_THRESHOLD,
   PIXELS_PER_LEVEL,
 } from '../../constants';
-import { TextAlign } from '../types';
+import { ClickedItemData, TextAlign } from '../types';
 
 import { FlameGraphDataContainer, LevelItem } from './dataTransform';
 
+const ufuzzy = new uFuzzy();
+
+export function useFlameRender(
```

Comment:
Mainly just moved and split to separate hooks but there weren't that many changes here.